temporal/server_test.go

```diff
@@ -37,16 +43,93 @@ import (
 	"go.temporal.io/server/tests/testutils"
 )
 
-// TestNewServer verifies that NewServer doesn't cause any fx errors
+// TestNewServer verifies that NewServer doesn't cause any fx errors, and that there are no unexpected error logs after
+// running for a few seconds.
 func TestNewServer(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	cfg := loadConfig(t)
+	logger := newErrorLogDetector(t, ctrl)
+
+	server, err := NewServer(
+		WithConfig(cfg),
+		WithLogger(logger),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		assert.NoError(t, server.Stop())
+	})
+	require.NoError(t, server.Start())
+	time.Sleep(10 * time.Second)
+}
+
+func loadConfig(t *testing.T) *config.Config {
+	cfg := loadSQLiteConfig(t)
+	setTestPorts(cfg)
+
+	return cfg
+}
+
+// loadSQLiteConfig loads the config for the sqlite persistence store. We use sqlite because it doesn't require any
+// external dependencies, so it's easy to run this test in isolation.
+func loadSQLiteConfig(t *testing.T) *config.Config {
 	configDir := path.Join(testutils.GetRepoRootDirectory(), "config")
 	cfg, err := config.LoadConfig("development-sqlite", configDir, "")
 	require.NoError(t, err)
+
 	cfg.DynamicConfigClient.Filepath = path.Join(configDir, "dynamicconfig", "development-sql.yaml")
-	_, err = NewServer(
-		ForServices(DefaultServices),
-		WithConfig(cfg),
-	)
-	assert.NoError(t, err)
-	// TODO: add tests for Server.Run(), etc.
+
+	return cfg
+}
+
+// setTestPorts sets the ports of all services to something different from the default ports, so that we can run the
+// tests in parallel.
+func setTestPorts(cfg *config.Config) {
+	port := 10000
+
+	for k, v := range cfg.Services {
+		rpc := v.RPC
+		rpc.GRPCPort = port
+		port++
+		rpc.MembershipPort = port
+		port++
+		v.RPC = rpc
+		cfg.Services[k] = v
+	}
+}
+
+// newErrorLogDetector returns a logger that fails the test if it logs any errors or warnings, except for the ones that
+// are expected. Ideally, there are no "expected" errors or warnings, but we still want this test to avoid introducing
+// any new ones while we are working on removing the existing ones.
+func newErrorLogDetector(t *testing.T, ctrl *gomock.Controller) *log.MockLogger {
+	logger := log.NewMockLogger(ctrl)
+	logger.EXPECT().Debug(gomock.Any(), gomock.Any()).AnyTimes()
+	logger.EXPECT().Info(gomock.Any(), gomock.Any()).AnyTimes()
+	logger.EXPECT().Warn(gomock.Any(), gomock.Any()).AnyTimes().Do(func(msg string, tags ...tag.Tag) {
+		if strings.Contains(msg, "error creating sdk client") {
```

Comment:
I'm thinking about how to handle this better... what I'd ideally like is for the test to set `sdkClientFactoryRetryExpirationInterval` to 9s so we make sure we can connect. but it seems overkill to add config just for this test (it's currently a hardcoded constant). can we check that we don't get more than three of these and fail if we see a fourth? that'll give it an effective timeout of 1400ms

Reply:
I'm a little apprehensive to do this because it could be ok for it to fail over and over and then eventually recover somewhere between 10s and 60s

Reply:
I think it's okay in the test environment.. it should connect as soon as the frontend starts its listener and ringpop stabilizes. You can even make the limit 5 calls (effective timeout of 6.2s plus some jitter, since initial interval is 200ms).

It was just a thought though.. other tests will fail if the worker can't connect so we don't have to do it here. Though that's using the onebox fx stuff and this is using the real version so maybe there still is some value